packages/block-library/src/term-template/edit.js

```diff
@@ -0,0 +1,512 @@
+			termsById[ term.parent ].children.push( termsById[ term.id ] );
+		} else {
+			rootTerms.push( termsById[ term.id ] );
+		}
+	} );
+
+	return rootTerms;
+}
+
+/**
+ * Renders a single term node and its children recursively.
+ *
+ * @param {Object}   termNode    Term node with term object and children.
+ * @param {Function} renderTerm  Function to render individual terms.
+ * @param {string}   blockLayout Layout type ('list' or 'grid').
+ * @return {JSX.Element} Rendered term node with children.
+ */
+function renderTermNode( termNode, renderTerm, blockLayout = 'list' ) {
+	const ContainerElement = blockLayout === 'grid' ? 'div' : 'ul';
+
+	return (
+		<>
+			{ renderTerm( termNode.term ) }
+			{ termNode.children.length > 0 && (
+				<ContainerElement>
+					{ termNode.children.map( ( child ) =>
+						renderTermNode( child, renderTerm, blockLayout )
+					) }
+				</ContainerElement>
+			) }
+		</>
+	);
+}
+
+/**
+ * Checks if a term is the currently active term.
+ *
+ * @param {number} termId               The term ID to check.
+ * @param {number} activeBlockContextId The currently active block context ID.
+ * @param {Array}  blockContexts        Array of block contexts.
+ * @return {boolean} True if the term is active, false otherwise.
+ */
+function isActiveTerm( termId, activeBlockContextId, blockContexts ) {
+	return termId === ( activeBlockContextId || blockContexts[ 0 ]?.termId );
+}
+
+export default function TermTemplateEdit( {
+	clientId,
+	attributes,
+	setAttributes,
+	context: {
+		termQuery: {
+			taxonomy,
+			order,
+			orderBy,
+			hideEmpty,
+			hierarchical,
+			parent,
+			perPage = 100,
+		} = {},
+	},
+} ) {
+	const [ activeBlockContextId, setActiveBlockContextId ] = useState();
+	const { replaceInnerBlocks } = useDispatch( blockEditorStore );
+
+	const queryArgs = {
+		order,
+		orderby: orderBy,
+		hide_empty: hideEmpty,
+		per_page: perPage,
+	};
+
+	const { records: terms, isResolving } = useEntityRecords(
+		'taxonomy',
+		taxonomy,
+		queryArgs
+	);
+
+	// Filter to show only top-level terms if "Show only top-level terms" is enabled.
+	const filteredTerms = useMemo( () => {
+		if ( ! terms || parent !== 0 ) {
+			return terms;
+		}
+		return terms.filter( ( term ) => ! term.parent );
+	}, [ terms, parent ] );
+
+	const { blocks, variations, defaultVariation } = useSelect(
+		( select ) => {
+			const { getBlocks } = select( blockEditorStore );
+			const { getBlockVariations, getDefaultBlockVariation } =
+				select( blocksStore );
+
+			return {
+				blocks: getBlocks( clientId ),
+				variations: getBlockVariations( 'core/term-template', 'block' ),
+				defaultVariation: getDefaultBlockVariation(
+					'core/term-template',
+					'block'
+				),
+			};
+		},
+		[ clientId ]
+	);
+
+	const blockLayout = attributes?.blockLayout || 'list';
+	const columnCount = attributes?.columnCount || 3;
+
+	// Switch to list if hierarchical is true and grid is selected.
+	useMemo( () => {
```

Comment:
Did you mean to use `useEffect()` instead of `useMemo()` here?

Reply:
Gah, yes, `useEffect` would make more sense here for updating attributes. Fixed in https://github.com/WordPress/gutenberg/pull/70720/commits/208ab8c4443a51796d8bab7b22b6b6e8dd09f611. Thanks!